core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -97,6 +102,13 @@ class Replica(val brokerId: Int, val topicPartition: TopicPartition) extends Log
     trace(s"Updated HW of replica to $highWatermark")
   }
 
+  def updateReassignmentLag(messageOffset: Long, isAddingReplica: Boolean): Unit = {
+    if (isAddingReplica)
```

Comment:
Given that we always call this with `updateLastSentHighWatermark`, does it make sense to put the logic there?

Reply:
I'm a bit torn in this case. I'm not sure either that the current solution can be defendable as you point out there is a dependency between these variables. So when I update the HW I probably want to update the lag as well, however I don't know if an `updateLastSentHighWatermark(hw: Long, lag: Long, isAdding: Boolean)` would be a good function. Perhaps it'd be better to update these parameters separately and compute the lag inside this method. Let me try that out.